Marlin/src/inc/Conditionals_post.h

```diff
@@ -1733,11 +1733,11 @@
 // When sharing the SD card with a PC we want the menu options to
 // mount/unmount the card and refresh it. So we disable card detect.
 //
-#if ENABLED(USB_SD_ONBOARD)
+#if ALL(USB_SD_ONBOARD, SDSUPPORT, LPC_SD_ONBOARD)
   #undef SD_DETECT_PIN
   #define SHARED_SD_CARD
 #endif
```

Comment:
Pointless change. `USB_SD_ONBOARD` cannot be enabled without `LPC_SD_ONBOARD` and `LPC_SD_ONBOARD` cannot be enabled without `SDSUPPORT`.

I'm just bringing this over from the pins files and not trying to change any behavior in this set of lines. Nothing controversial to see here.

Reply:
I don't agree with this ,USB_SD_ONBOARD can be set without LPC_SD_ONBOARD, it is the default setting here in the SKR V1.3 pins file...
```C
#if !ANY(LPC_SD_LCD, LPC_SD_ONBOARD, LPC_SD_CUSTOM_CABLE)
  #undef USB_SD_DISABLED
  #define USB_SD_ONBOARD
  #define LPC_SD_LCD
#endif
```
This configuration allows for the onboard card to be accessed via USB (for firmware updates) but uses the LCD SD card for Marlin. In that configuration the SD card is not being shared and so SHARED_SD_CARD should not be set and the SD_DETECT_PIN should be left alone. Adding the extra SDSUPPORT test does no harm and in my view correctly describes the conditions fpr a shared SD card.

Reply:
Scott have you seen my comments above. My testing has shown that the code you have reverted this back to is not correct. It does not correctly handle a common case. The code you moved from the pins file was not stand alone it was in the else part of a test on LPC_SD_LCD (which is mutually exclusive to to LPC_SD_ONBOARD - no I don't like this defines very much either)....
original pins file code...
```C
#if ENABLED(LPC_SD_LCD)

  #define SCK_PIN          P0_15
  #define MISO_PIN         P0_17
  #define MOSI_PIN         P0_18
  #define SS_PIN           P0_16   // Chip select for SD card used by Marlin
  #define ONBOARD_SD_CS    P0_06   // Chip select for "System" SD card

#elif ENABLED(LPC_SD_ONBOARD)

  #if ENABLED(USB_SD_ONBOARD)
    // When sharing the SD card with a PC we want the menu options to
    // mount/unmount the card and refresh it. So we disable card detect.
    #define SHARED_SD_CARD
    #undef SD_DETECT_PIN
    //#define SD_DETECT_PIN  P0_27   // (57) open-drain
  #endif
```
As you can see an extra test is needed either on LPC_SD_ONBOARD or on LPC_SD_LCD.